### PR TITLE
Feature ajax

### PR DESCRIPTION
**Pull request by mixcas**
Add ajax functionality, transitions and rebindings.

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
 
 };
 
+// AJAX
+Ajaxy = {
+  init: function() {
+    var _this = this;
+    var ajaxyLinks = 'a.ajax-link';
```
Comment (varffs) [inline js/main.js]: why you declaring the links like this?

File: js/main.js [hunk 2]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
...
+
+    // For back button
+    window.onpopstate = function() {
+      _this.ajaxLoad(document.location.origin + document.location.pathname);
+    };
+  },
+
+  reset: function() {
+    var _this = this;
+    $('a.ajax-link').unbind('click');
```
Comment (varffs) [inline js/main.js]: and then not using a variable at all here?

File: js/main.js [hunk 3]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
...
+
+      dataType: 'html',
+      error: function(jqXHR, textStatus) {
+        _this.ajaxErrorHandler(jqXHR, textStatus);
+      },
+
+      success: function(data) {
+        _this.ajaxSuccess(data, url);
+        setTimeout( function() {
+        }, animationSpeed);
```
Comment (varffs) [inline js/main.js]: what is this?

File: js/main.js [hunk 4]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
...
+
+      complete: function() {
+        _this.ajaxAfter();
+      },
+    });
+  },
+
+  ajaxBefore: function() {
+    $('.nav').addClass('nav-hidden');
+    $('#main-container').addClass('main-hidden');
```
Comment (varffs) [inline js/main.js]: should these constant elements be cached?

File: js/main.js [hunk 5]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
...
+    // Push new history state and update title
+    history.pushState(null, $title, url);
+    document.title = $title;
+
+    // Update with new content and classes
+    $('#main-container').html($content.html());
+    $('body').removeAttr('class').addClass($bodyClasses);
+
+    if( $bodyClasses.indexOf('home') !== 0 ) {
+      // TODO: Check if threejs is running, if not, init it.
```
Comment (varffs) [inline js/main.js]: why leave this todo in the PR? Is this not part of the ajax feature?

File: js/main.js [hunk 6]
```diff
@@ -411,6 +436,109 @@ TrophyModern.Speech = {
 
 };
 
+// AJAX
+Ajaxy = {
```
Comment (varffs) [inline js/main.js]: Should Ajaxy be a child of the TrophyModern object?

Comment (mixcas) [inline js/main.js]: yeah, why not?

**Pull request merged**